src/main/java/sparksoniq/spark/DataFrameUtils.java

```diff
@@ -225,24 +225,25 @@ public static String getGroupbyProjectSQL(
             }
             if (applyCount) {
                 queryColumnString.append("count(`");
+            } else if (applyDistinct) {
+                queryColumnString.append(serializerUdfName);
+                queryColumnString.append("(");
+                queryColumnString.append("first(");
             } else {
                 queryColumnString.append(serializerUdfName);
                 queryColumnString.append("(");
-                if (applyDistinct) {
```

Comment:
I think it might be nice to rename this variable to something else since we're not using a distinct operation anymore.